driver-sync/src/main/com/mongodb/client/internal/ClientSessionImpl.java

```diff
@@ -154,12 +158,21 @@ public void abortTransaction() {
                         readConcern, this);
             }
         } catch (Exception e) {
-            // ignore errors
+            if (e instanceof MongoException) {
+                unpinServerAddressOnError((MongoException) e);
+            }
         } finally {
             cleanupTransaction(TransactionState.ABORTED);
         }
     }
 
+    private void unpinServerAddressOnError(final MongoException e) {
+        if (getPinnedServerAddress() != null
```

Comment:
Nit: It's harmless to unpin an already-unpinned server, so you can simplify the conditional by removing the null check.

Reply:
I have removed the null check.